### PR TITLE
[GLUTEN-1945] Support summarizing the supported spark built-in functions

### PR DESCRIPTION
**Pull request by PHILO-HE**
This is a rough summary by testing example query for spark registered functions.
There are some false positive & false negative cases for some functions. For such situation, we need to use a suitable test sql to do the check. Will include this in a follow-up PR.

### CODE REVIEW

File: gluten-ut/spark32/src/test/scala/org/apache/spark/sql/statistics/SparkFunctionStatistics.scala [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+          }
+          }
+          if (isSupported && !needInspectFunctions.contains(exprInfo.getName)) {
+            supportedFunctions.add(exprInfo.getName)
+          } else if (!isSupported && !needInspectFunctions.contains(exprInfo.getName)) {
+            unsupportedFunctions.add(exprInfo.getName)
+          }
+      }
+    }
+    print("overall functions: " + (sparkBuiltInFunctions.size - ignoreFunctions.size) + "\n")
```
Comment (pan3793) [inline gluten-ut/spark32/src/test/scala/org/apache/spark/sql/statistics/SparkFunctionStatistics.scala:129]: why not `println`?

Comment (PHILO-HE) [reply gluten-ut/spark32/src/test/scala/org/apache/spark/sql/statistics/SparkFunctionStatistics.scala]: Fixed. Thanks!

File: gluten-ut/spark33/src/test/scala/org/apache/spark/sql/statistics/SparkFunctionStatistics.scala [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+              isSupported = false
+              break
+            }
+            break
+          }
+        }
+        if (isSupported && !needInspectFunctions.contains(exprInfo.getName)) {
+          supportedFunctions.add(exprInfo.getName)
+        } else if (!isSupported && !needInspectFunctions.contains(exprInfo.getName)) {
+          unsupportedFunctions.add(exprInfo.getName)
```
Comment (rui-mo) [inline gluten-ut/spark33/src/test/scala/org/apache/spark/sql/statistics/SparkFunctionStatistics.scala:139]: Is there any case meet this condition `!isSupported && needInspectFunctions.contains(exprInfo.getName)`? If not, this clause can just be like `else if (!isSupported)`.

Comment (PHILO-HE) [reply gluten-ut/spark33/src/test/scala/org/apache/spark/sql/statistics/SparkFunctionStatistics.scala:139]: Just removed the extra check condition that never goes into. Thanks!

File: gluten-ut/spark32/src/test/scala/org/apache/spark/sql/statistics/SparkFunctionStatistics.scala [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+              break
+            }
+            break
+          }
+        }
+        if (isSupported && !needInspectFunctions.contains(exprInfo.getName)) {
+          supportedFunctions.add(exprInfo.getName)
+        } else if (!isSupported) {
+          unsupportedFunctions.add(exprInfo.getName)
+        }
```
Comment (rui-mo) [inline gluten-ut/spark32/src/test/scala/org/apache/spark/sql/statistics/SparkFunctionStatistics.scala:140]: Does it make sense to add an unreachable check here for `else`?

Comment (PHILO-HE) [reply gluten-ut/spark32/src/test/scala/org/apache/spark/sql/statistics/SparkFunctionStatistics.scala:140]: The unchecked else is actually handled for `needInspectFunctions` in the below. Thanks!

**Pull request merged**